Mage.Sets/src/mage/cards/g/GildedDrake.java

```diff
@@ -72,24 +72,23 @@ public GildedDrakeEffect copy() {
 
     @Override
     public boolean apply(Game game, Ability source) {
-
         Player controller = game.getPlayer(source.getControllerId());
-        if (controller != null) {
-            Permanent sourceObject = game.getPermanent(source.getSourceId());
-            if (sourceObject != null) {
-                if (targetPointer.getFirst(game, source) != null) {
-                    Permanent targetPermanent = game.getPermanent(targetPointer.getFirst(game, source));
-                    if (targetPermanent != null) {
-                        ContinuousEffect effect = new ExchangeControlTargetEffect(Duration.EndOfGame, "", true);
-                        effect.setTargetPointer(targetPointer);
-                        game.addEffect(effect, source);
-                        return true;
-                    }
-                }
-                sourceObject.sacrifice(source, game);
-            }
+        if (controller == null) {
+            return false;
+        }
+        Permanent sourceObject = game.getPermanent(source.getSourceId());
+        if (sourceObject == null) {
```

Comment:
Small nitpick: I'd assign values to both controller and sourceObject and then check null for both variables in one if statement

Reply:
I personally like this approach, but this is more of a personal preference. I don't think either is especially better.

If you or someone else has a good reason to do it one way or another I'll switch it.

Reply:
Technically, your way might run slightly faster as it fails faster on null controller. I like my way because it is fewer lines but still clear what is going on. Not a big deal if you prefer your way though